src/runs/compiler.ts

```diff
@@ -19,13 +19,15 @@ export const getBinSaveLocation = (srcPath: string): string => {
     if (language.skipCompile) {
         return srcPath;
     }
+    const ext = language.name == 'java' ? '*.class' : '.bin';
     const savePreference = getSaveLocationPref();
-    const srcFileName = path.basename(srcPath);
-    const binFileName = `${srcFileName}.bin`;
+    const srcFileName = path.parse(srcPath).name;
+    const binFileName = srcFileName + ext;
+    const binDir = path.dirname(srcPath);
     if (savePreference && savePreference !== '') {
         return path.join(savePreference, binFileName);
     }
-    return `${srcPath}.bin`;
+    return path.join(binDir, binFileName);
```

Comment:
I have done this to change the binary/class (in case of Java) file name format from `problem_name.ext.bin/class` to `problem_name.bin/class`.